### PR TITLE
Use spf13/cobra for docker commit

### PR DESCRIPTION
**Pull request by yongtang**
This fix is part of the effort to convert commands to spf13/cobra #23211.

Thif fix coverted command `docker commit` to use spf13/cobra

Signed-off-by: Yong Tang yong.tang.github@outlook.com

### CODE REVIEW

File: integration-cli/docker_cli_commit_test.go [hunk 1]
```diff
@@ -116,9 +116,9 @@ func (s *DockerSuite) TestCommitChange(c *check.C) {
 		"--change", "ENV test 1",
 		"--change", "ENV PATH /foo",
 		"--change", "LABEL foo bar",
-		"--change", "CMD [\"/bin/sh\"]",
+		"--change", `"CMD [""/bin/sh""]"`,
```
Comment (LK4D4) [inline integration-cli/docker_cli_commit_test.go]: double quotes is really weird. is it like things escaped in cobra or what?

Comment (yongtang) [inline integration-cli/docker_cli_commit_test.go]: Thanks @LK4D4 for the review. It looks like cobra requires quotes from outside (as there is a space in `CMD` and `[]`. But then to escape the quotes inside the quotes, a double quotes will be needed.

Therefore the whole argument has to be (single quotes outside and double quotes inside):
```
"CMD [""/bin/sh""]"
```

Otherwise it will return error:
```
invalid argument "CMD [\"/bin/sh\"]" for --change: line 1, column 5: bare " in non-quoted-field
```

Comment (vdemeester) [inline integration-cli/docker_cli_commit_test.go]: ping @dnephin 
In a shell (`zsh`), with docker 1.11 : `λ docker commit --change 'CMD ["/bin/sh"]' 830` works fine, with current master `docker commit --change 'CMD ["/bin/sh"]' ef9` returns *"invalid argument "CMD [\"/bin/sh\"]" for --change: line 1, column 5: bare " in non-quoted-field"*.

File: api/client/container/commit.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+}
+
+// NewCommitCommand creats a new cobra.Command for `docker commit`
+func NewCommitCommand(dockerCli *client.DockerCli) *cobra.Command {
+	var opts commitOptions
+
+	cmd := &cobra.Command{
+		Use:   "commit [OPTIONS] CONTAINER [REPOSITORY[:TAG]]",
+		Short: "Create a new image from a container's changes",
+		Args:  cli.RequiresMinMaxArgs(1, 2),
```
Comment (dnephin) [inline api/client/container/commit.go]: `RequiresRangeArgs(1, 2)` ?

I guess this is already in master

File: integration-cli/docker_cli_commit_test.go (hunk 1 above)
Comment (dnephin) [inline integration-cli/docker_cli_commit_test.go]: Maybe related to https://github.com/dnephin/cobra/blob/fork/command.go#L358-L396 ?

I think the error is coming from https://golang.org/pkg/encoding/csv/, which I think is called from `opts.ListOpts`. I think pflags must have been doing something funky to make this work? Maybe the "add single quotes" patch that we didn't add.

What if you use a `StringSliceVar()` here instead of `ListOps` ?

File: api/client/container/commit.go (hunk 2 above)
Comment (yongtang) [inline api/client/container/commit.go]: Thanks @dnephin. `RequiresMinMaxArgs()` was added as part of the #23286. Now I have changed the name to `RequiresRangeArgs()` and updated the pull request.

**Comment by thaJeztah** [comment]
needs a rebase as well now

**Comment by yongtang** [comment]
Thanks @thaJeztah the PR has been rebased.

File: integration-cli/docker_cli_commit_test.go (hunk 1 above)
Comment (yongtang) [inline integration-cli/docker_cli_commit_test.go]: @dnephin At the moment `--change` use `StringSliceVarP`. I searched the web and most of the same error message `bare " in non-quoted-field"` points to golang's `csv` package. I don't know enough detail about the implementation in cobra though.

Comment (dnephin) [inline integration-cli/docker_cli_commit_test.go]: Oh, sorry I got that backwards. It used to be a ListOpts, so I think that
is the difference. Does it work if you go back to using the list opts, or
is it broken with both types ?

Comment (dnephin) [inline integration-cli/docker_cli_commit_test.go]: Ya, `StringSliceVar` is the one that uses csv, and `ListOpts` does not. So changing the type back to the old type should fix this issue and you'll be able to revert this change to the test.

Comment (yongtang) [inline integration-cli/docker_cli_commit_test.go]: @dnephin I tried with `ListOpts` and it woks as expected. Thanks for the help!

The pull request has been updated. Please let me know if there are any issues.

**Comment by yongtang** [comment]
Thanks @dnephin for the help. The pull request has been updated. Please let me know if there are any issues.

**Comment by dnephin** [comment]
LGTM

**Comment by LK4D4** [comment]
LGTM
